### PR TITLE
Update coil100 dataset

### PR DESCRIPTION
**Pull request by harsh020**
## Update coil100 dataset

Fixes issue #2745 

Tasks:
* [x] Rename `label` to `angle_label`
* [x] Convert `object_id` to `ClassLabel`
* [x] Create `angle` key for `raw_angles` of `tf.int64` dtype.  
* [x] Update documentations

### CODE REVIEW

**Comment by harsh020** [comment]
@Conchylicultor Hi, currently, I have just committed the changes where I have just renamed `label` to `angle_label`. It is giving the error that I mentioned [here](https://github.com/tensorflow/datasets/issues/2745#issuecomment-731393146).

The tasks of - 
* Convert object_id to ClassLabel
* Create angle key for raw_angles of tf.int64 dtype.
Also raise the same error.

Thank You.

File: docs/catalog/coil100.md [hunk 1]
```diff
@@ -44,12 +44,13 @@ Split  | Examples
 ```python
```
Comment (Conchylicultor) [inline docs/catalog/coil100.md:44]: Doc is automatically generated

**Review by Conchylicultor — APPROVED** [review]
Thank you.

Note: `Version` should be updated and RELEASE_NOTES added when updating a dataset

File: tensorflow_datasets/image/coil100.py [hunk 2]
```diff
@@ -85,10 +87,12 @@ def _generate_examples(self, data_dir_path):
     for file_name in tf.io.gfile.listdir(data_dir_path):
       if file_name.endswith(".png"):
         image = os.path.join(data_dir_path, file_name)
-        label = file_name.split("_")[2].split(".")[0]
+        angle_label = file_name.split("_")[2].split(".")[0]
+        angle = tf.cast(int(angle_label), tf.int64)
```
Comment (Conchylicultor) [inline tensorflow_datasets/image/coil100.py:91]: Did you actually tested this code ? _generate_examples should yield np.array compatible values, not tf.Tensor.

Comment (harsh020) [reply tensorflow_datasets/image/coil100.py:91]: I tested the updated dataset with `load` and it seem to be yielding each example as a `Dict`. 
The signature of each example is as -
```
{
  'angle': <tf.Tensor 'args_0:0' shape=() dtype=int64>, 
  'angle_label': <tf.Tensor 'args_1:0' shape=() dtype=int64>, 
  'image': <tf.Tensor 'args_2:0' shape=(128, 128, 3) dtype=uint8>, 
  'object_id': <tf.Tensor 'args_3:0' shape=() dtype=int64>
}
```

Thank you.

Comment (harsh020) [reply tensorflow_datasets/image/coil100.py]: Ok! I think you mean instead of `tf.cast` we should use `numpy`conversion for `np.array` compatibility.

**Comment by harsh020** [comment]
@Conchylicultor Have resolved the merge conflict, kindly merge into master as currently the master does not contain `angle` feature.

Thank you.

**Pull request closed without merging**